File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        mapi ={ "]":"[","}":"{",")":"("}
        stack=[]
        for i in s:
            if i in mapi:
                top=stack.pop() if stack else "#"
                if mapi[i] != top:
                    return False
        
            else:
                stack.append(i)
        return not stack

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<>();
        char[] c = s.toCharArray();
        for(char i : c){
            if(i == '(' || i == '{' || i == '['){
                st.push(i);
            }else{
                if(st.isEmpty())return false;
                char x = st.pop();
                
                if( i == ')' && x != '('){
                    return false;
                }else if ( i == '}' && x != '{'){
                    return false;
                }else if ( i == ']' && x != '['){
                    return false;
                }
            }
        }
        return true ? st.isEmpty() : false;  
        
    }
}